### ISSUE TITLE
Add TrieView::get(&self) -> Option<&V> Functionality

### ISSUE THREAD

**Issue by Kylebrown9** [Randevelopment/Slang#6]
The intention was for `Trie::get()` to optionally return a value, not a view.
A method called `Trie::get_view()` would be a good idea, and a good way to implement the intended `get` behavior, but is a distinct concept.

**Comment by Kylebrown9** [comment]
In the long run the solution I really want for most of these problems is to use Generic Associated Types for TrieView::View and TrieView::ViewMut to incorporate lifetimes.

**Comment by KrishnaSannasi** [comment]
Yes, GATs would solve this. Last I checked, they were going to implement Generic Associated Lifetimes, a subset of GATs that only allow parameterizing associated types on lifetimes. This would also solve many issues.

**Comment by Kylebrown9** [comment]
I've been trying to look into the GAT's situation and the best estimate I've gotten is

"last I looked into it I would expect a stable GAT timeline to be on the order of months"

**Comment by KrishnaSannasi** [comment]
Yes, that sounds about right

**Comment by KrishnaSannasi** [comment]
I just remembered seeing a blog post on how to emulate GATs in today's Rust.

example
```rust
trait NeedsGat {
    type Getter<T>;
    fn get<T>(g: Self::Getter<T>) -> T;
}
```
Can be turned into


```rust
trait Gat<T> {
    type Getter;
}
trait NeedsGat {
    fn get<T>(g: Self::Getter) -> T where Self: Gat<T>;
}
```

which effectively emulates GATs.

For our case, we could use
```rust
pub trait HasView<'a> {
    type View;
}

pub trait TrieView {
    fn view<'a>(&'a self) -> Self::View where Self: HasView<'a>;
}
```

Which would give us associated lifetimes for only a little boilerplate!

**Comment by KrishnaSannasi** [comment]
This is using the idea that traits are type level functions, generic and lifetime parameters are inputs to the type level functions and associated types are the outputs of these type level functions.

**Comment by Kylebrown9** [comment]
Resolved by pull-request #11 "Issue 6"
